contrib/go/examples/src/go/distance/BUILD

```diff
@@ -0,0 +1,11 @@
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# This target must be build with protoc that recognizes `option go_package`
+# ./pants run contrib/go/examples/src/go/distance/ --protoc-version=3.5.1
```

Comment:
`option go_package` does not appear to be available in protobuf 2.x, which is what the Pants build currently defaults to. I'm not sure how to deal with this, other than change the protoc version at runtime.

One idea that jumps to mind is tagging this target to indicate that it requires proto3, and we could have a separate CI job that runs with proto3. I'm totally open to suggestions for how to deal with this.

Reply:
Generally targets that do not build without extra flags end up needing to be defined under `<root>/testprojects`.

But because `golang` doesn't support multiple sourceroots, I think what you might need to do here is define this target in a file like `TEST_BUILD`, and then rename it for the purpose of an integration test. See https://github.com/pantsbuild/pants/blob/master/tests/python/pants_test/backend/python/test_pants_requirement_integration.py#L36-L46 for an example of doing this.

But please do add an integration test that covers building the binary.

Reply:
Could you update the comment for the more specific protoc arg?